### ISSUE TITLE
Editor crashes upon entity creation after updating fbx settings in Procedural Prefab

### ISSUE THREAD

**Issue by LB-BartoszDudzinski** [o3de/o3de#7197]
**Describe the bug**
The Asset Processor fails to update fbx settings in a Procedural Prefab asset. Creating an entity afterwards crashes the Editor.

The issue also occurs on Linux.

**Steps to reproduce**
1. Create a level.
2. In the Asset Browser search for a fbx Procedural Prefab.
3. Press the RMB on it and select "Edit Settings".
4. Modify the fbx and press "Update".
5. Close the fbx settings and create an entity.

**Expected behavior**
The Editor remains stable.

**Actual behavior**
The Editor crashes.

**Call Stack Trace:**
```
25) 00007FFBA90EB857 (KERNELBASE) : UnhandledExceptionFilter
24) 00007FFBAB8151B0 (ntdll) : memset
23) 00007FFBAB7FC766 (ntdll) : _C_specific_handler
22) 00007FFBAB8120CF (ntdll) : _chkstk
21) 00007FFBAB7C1454 (ntdll) : RtlRaiseException
20) 00007FFBAB810BFE (ntdll) : KiUserExceptionDispatcher
19) C:\o3de\o3de-packages\packages\RapidJSON-1.1.0-rev1-multiplatform\RapidJSON\include\rapidjson\allocators.h (321) : rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator>::Malloc
18) C:\o3de\o3de-packages\packages\RapidJSON-1.1.0-rev1-multiplatform\RapidJSON\include\rapidjson\document.h (745) : rapidjson_ly::GenericValue<rapidjson_ly::UTF8<char>,rapidjson_ly::MemoryPoolAllocator<rapidjson_ly::CrtAllocator> >::GenericValue<rapidjson_l
17) C:\o3de\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonMerger.cpp (462) : AZ::JsonMerger::ApplyPatch_Replace
16) C:\o3de\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonMerger.cpp (93) : AZ::JsonMerger::ApplyPatch
15) C:\o3de\o3de\Code\Framework\AzCore\AzCore\Serialization\Json\JsonSerialization.cpp (128) : AZ::JsonSerialization::ApplyPatch
14) C:\o3de\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\PrefabDomUtils.cpp (362) : AzToolsFramework::Prefab::PrefabDomUtils::ApplyPatches
13) C:\o3de\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\Instance\InstanceToTemplatePropagator.cpp (168) : AzToolsFramework::Prefab::InstanceToTemplatePropagator::PatchTemplate
12) C:\o3de\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\PrefabUndoHelpers.cpp (30) : AzToolsFramework::Prefab::PrefabUndoHelpers::UpdatePrefabInstance
11) C:\o3de\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\Prefab\PrefabPublicHandler.cpp (697) : AzToolsFramework::Prefab::PrefabPublicHandler::CreateEntity
10) C:\o3de\o3de\Code\Framework\AzToolsFramework\AzToolsFramework\UI\Prefab\PrefabIntegrationManager.cpp (1358) : AzToolsFramework::Prefab::PrefabIntegrationManager::CreateNewEntityAtPosition
 9) C:\o3de\o3de\Code\Editor\Plugins\ComponentEntityEditorPlugin\SandboxIntegration.cpp (1303) : SandboxIntegrationManager::CreateNewEntityAtPosition
 8) C:\o3de\o3de-packages\packages\qt-5.15.2-rev4-windows\qt\include\QtCore\qobjectdefs_impl.h (443) : QtPrivate::QFunctorSlotObject<`SandboxIntegrationManager::PopulateEditorGlobalContextMenu'::`19'::<lambda_1>,0,QtPrivate::List<>,void>::impl
 7) 00007FFB659C0DB9 (Qt5Core) : QObject::qt_static_metacall
 6) 00007FFB6F195DCF (Qt5Widgets) : QAction::activate
 5) 00007FFB6F2EC698 (Qt5Widgets) : QMenu::actionGeometry
 4) 00007FFB6F2EC473 (Qt5Widgets) : QMenu::actionGeometry
 3) 00007FFB6F2F1C51 (Qt5Widgets) : QMenu::mouseReleaseEvent
 2) 00007FFB6F1C7B02 (Qt5Widgets) : QWidget::event
 1) 00007FFB6F2EDC37 (Qt5Widgets) : QMenu::event
```

**Screenshots/Video**

https://user-images.githubusercontent.com/86952009/151368612-688ed121-9708-4937-9cdd-fd6d8038932b.mp4

**Found in Branch**
Development (https://github.com/o3de/o3de/commit/e2c4c0e5e3e0b7bd4a3adf2fe0ffee6c9b454483)

**Desktop/Device:**
- Device: PC
- OS: Windows
- Version 10
- CPU Intel Core i5-9600k
- GPU Nvidia GeForce RTX 2060
- Memory 16GB

**Additional context**
[Editor.log](https://github.com/o3de/o3de/files/7950795/Editor.log)
[error.log](https://github.com/o3de/o3de/files/7950796/error.log)
[dump.zip](https://github.com/o3de/o3de/files/7950797/dump.zip)

**Comment by LesaelR** [comment]
Verified as Fixed in Development f5fcab75d6

**Comment by LB-BartoszDudzinski** [comment]
Verified as Fixed in Development.
